packages/backend/server/src/core/version/service.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable, Logger } from '@nestjs/common';
+import semver from 'semver';
+
+import { Runtime, UnsupportedClientVersion } from '../../base';
+
+@Injectable()
+export class VersionService {
+  private readonly logger = new Logger(VersionService.name);
+
+  constructor(private readonly runtime: Runtime) {}
+
+  private async getVersionRange(versionRange: string) {
+    let range: semver.Range | undefined;
+    try {
+      range = new semver.Range(versionRange, { loose: false });
```

Comment:
Do we need to add a cache? I'm not sure if the performance of creating a large number of semver.Range instances will be affected, as I learned a long time ago that the performance of the semver package is not very good.